### PR TITLE
parts: allow cleaning specific parts

### PR DESCRIPTION
**Pull request by cmatsuoka**
The clean command can now clean specific parts inside the managed environment.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: rockcraft/parts.py [hunk 1]
```diff
@@ -76,6 +76,20 @@ def __init__(
         except craft_parts.PartsError as err:
             raise PartsLifecycleError(str(err)) from err
 
+    def clean(self) -> None:
+        """Remove lifecycle artifacts.
+
+        :param part_names: The names of the parts to clean. If not
```
Comment (tigarmo) [inline rockcraft/parts.py:82]: feels a bit awkward to have this `:param` here if the method doesn't take parameters

Comment (cmatsuoka) [reply rockcraft/parts.py:82]: Whoops. Deleted the argument and documentation remained.

**Review by tigarmo — APPROVED** [review]
thanks!

**Review by mr-cal — APPROVED** [review]
+1 for the PR and +1 for resolving TODOs that I created = +2

**Pull request merged**